codegen-v2/src/main/kotlin/club/doki7/babel/cdecl/tokenize.kt

```diff
@@ -187,82 +180,19 @@ class CTokenizer(private val source: List<String>, var curLine: Int) {
-                if (!noTransform && curToken.value.isLikelyMacro() && pos + 1 < tokens.size
-                    && tokens[pos + 1].type == RawTokenType.SYMBOL
-                    && tokens[pos + 1].value == "("
-                ) {
-                    // macro as function
-                    val sb = StringBuilder()
-                    sb.append(tokens[pos].value) // macro name
-                    pos++
-                    sb.append(tokens[pos].value) // (
-                    if (pos + 1 >= tokens.size) {
-                        syntaxError("Unexpected end of tokens", tokens[pos].line, tokens[pos].col)
-                    }
-                    pos++
-                    when (tokens[pos].type) {
-                        RawTokenType.IDENT -> {
-                            sb.append(tokens[pos].value) // identifier
-                            pos++
-                        }
-
-                        RawTokenType.STRING -> {
-                            sb.append("\"")
-                            sb.append(tokens[pos].value) // string
-                            sb.append("\"")
-                            pos++
-                        }
-
-                        else -> {}
-                    }
-                    if (pos >= tokens.size) {
-                        syntaxError("Unexpected end of tokens", tokens.last().line, tokens.last().col)
-                    }
-                    if (tokens[pos].type != RawTokenType.SYMBOL || tokens[pos].value != ")") {
-                        syntaxError("Expected ')'", tokens[pos])
-                    }
-                    sb.append(tokens[pos].value) // )
-                    result.add(Token(TokenType.TRIVIA, sb.toString(), curToken.line, curToken.col))
-                } else if (!noTransform && curToken.value.isLikelyMacro()) {
-                    // macro without args
-                    result.add(Token(TokenType.TRIVIA, curToken.value, curToken.line, curToken.col))
-                } else if (isTypeQualifier(curToken.value)) {
-                    // type qualifier
-                    result.add(Token(TokenType.QUALIFIER, curToken.value, curToken.line, curToken.col))
-                } else if (isKeyword(curToken.value)) {
-                    // keyword
-                    result.add(Token(TokenType.KEYWORD, curToken.value, curToken.line, curToken.col))
-                } else {
-                    // normal identifier
-                    result.add(curToken.toToken())
-                }
-            }
-            else -> {
-                // other tokens
-                result.add(curToken.toToken())
-            }
-        }
-        pos++
-    }
-    return result
-}
-
-private val typeQualifiers = arrayOf(CONST, VOLATILE)
-
-private fun isTypeQualifier(s: String): Boolean = s in typeQualifiers
-
-private val keywords = arrayOf(TYPEDEF)
-
-private fun isKeyword(s: String): Boolean = s in keywords
-
+internal val knownMacros = setOf(
```

Comment:
Maybe we can really parse `#define` things and exactly know what are macros in the context.